AclHelper.cpp

```diff
@@ -0,0 +1,32 @@
+#include "pch.h"
+#include "AclHelper.h"
+
+AclHelper::AclHelper()
+{
+    emptyAcl = (PACL)malloc(sizeof(ACL));
+
+    if (emptyAcl)
+    {
+        InitializeAcl(emptyAcl, sizeof(ACL), ACL_REVISION);
+    }
+}
+
+AclHelper::~AclHelper()
+{
+    if (emptyAcl)
+    {
+        free(emptyAcl);
+    }
+}
+
+DWORD AclHelper::ResetAcl(const wstring& path)
+{
+    if (emptyAcl)
+    {
+        return SetNamedSecurityInfoW(const_cast<LPWSTR>(path.c_str()), SE_FILE_OBJECT, DACL_SECURITY_INFORMATION | UNPROTECTED_DACL_SECURITY_INFORMATION, NULL, NULL, emptyAcl, NULL);
```

Comment:
Once we changed the security information to what we need, shouldn't we restore the default security information ?

Reply:
We reset it, so it inherits what is on the Program Files folder, which is the default behavior.